ibm/service/kms/resource_ibm_kms_kmip_adapter.go

```diff
@@ -0,0 +1,255 @@
+				Description:      "Key protect Instance GUID",
+				ForceNew:         true,
+				DiffSuppressFunc: suppressKMSInstanceIDDiff,
+			},
+			"profile": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				Description:  "The profile of the KMIP adapter",
+				ValidateFunc: validate.ValidateAllowedStringValues(resourceIBMKmsKMIPAdapterValidProfiles),
+			},
+			"profile_data": {
+				Type:        schema.TypeMap,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The data specific to the KMIP Adapter profile",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "The name of the KMIP adapter",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The description of the KMIP adapter",
+			},
+			"created_by": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The unique identifier that is associated with the entity that created the adapter.",
+			},
+			"created_at": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date when a resource was created. The date format follows RFC 3339.",
+			},
+			"updated_by": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The unique identifier that is associated with the entity that updated the adapter.",
+			},
+			"updated_at": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date when a resource was updated. The date format follows RFC 3339.",
+			},
+		},
+	}
+}
+
+func resourceIBMKmsKMIPAdapterProfileToProfileFunc(profile string, profileData map[string]string) kp.CreateKMIPAdapterProfile {
+	if profile == resourceIBMKmsKMIPAdapterValidProfiles[0] {
+		//native_1.0
+		return kp.WithNativeProfile(profileData["crk_id"])
+	}
+	// Shouldn't reach here, since we check for profile validity before this.
+	return nil
+}
+
+func resourceIBMKmsKMIPAdapterCreate(d *schema.ResourceData, meta interface{}) error {
+	adapterToCreate, instanceID, err := ExtractAndValidateKMIPAdapterDataFromSchema(d)
+	if err != nil {
+		return err
+	}
+	kpAPI, _, err := populateKPClient(d, meta, instanceID)
+	if err != nil {
+		return err
+	}
+	adapter, err := kpAPI.CreateKMIPAdapter(context.Background(),
+		resourceIBMKmsKMIPAdapterProfileToProfileFunc(adapterToCreate.Profile, adapterToCreate.ProfileData),
+		kp.WithKMIPAdapterName(adapterToCreate.Name),
+		kp.WithKMIPAdapterDescription(adapterToCreate.Description),
+	)
+	if err != nil {
+		return fmt.Errorf("[ERROR] Error while creating KMIP adapter: %s", err)
+	}
+	return populateKMIPAdapterSchemaDataFromStruct(d, *adapter)
+}
+
+func resourceIBMKmsKMIPAdapterRead(d *schema.ResourceData, meta interface{}) error {
+	instanceID := d.Get("instance_id").(string)
```

Comment:
Can you set d.SetID as combination of instance_id and adapter_id separated by "/" 
We can't do d.Get() in read method. 
Import scenario breaks